src/gl_limits.rs

```diff
@@ -7,19 +7,33 @@ use serde::{Deserialize, Deserializer, Serialize, Serializer};
 pub struct GLLimits {
     pub max_vertex_attribs: u32,
     pub max_tex_size: u32,
-    pub max_cube_map_tex_size: u32
+    pub max_cube_map_tex_size: u32,
+    pub max_combined_texture_image_units: u32,
+    pub max_fragment_uniform_vectors: u32,
+    pub max_renderbuffer_size: u32,
+    pub max_texture_image_units: u32,
+    pub max_varying_vectors: u32,
+    pub max_vertex_texture_image_units: u32,
+    pub max_vertex_uniform_vectors: u32
 }
 
 #[cfg(feature = "serde")]
 impl<'de> Deserialize<'de> for GLLimits {
     fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
         where D: Deserializer<'de>
     {
-        let values = try!(<[_; 3]>::deserialize(deserializer));
+        let values = try!(<[_; 10]>::deserialize(deserializer));
```

Comment:
any reason why `GLLimits` can't just derive deserialization?

Reply:
This was an explicit choice a while ago in https://github.com/emilio/rust-offscreen-rendering-context/commit/c97823f97b8fbee8689d5cdaabb22566c03007be (from #57). I'd rather look into that separately from this work.